git-config/src/values.rs

```diff
@@ -1390,3 +1460,30 @@ mod color_attribute {
         assert!(ColorAttribute::from_str("no-").is_err());
     }
 }
+
+#[cfg(test)]
+mod path {
+    use super::{Path, TryFrom};
+    use std::borrow::Cow;
+    use std::path::Path as StdPath;
+    use tempfile::tempdir;
+    use std::fs::File;
+
+    #[test]
+    fn ok() {
+        let dir = tempdir().unwrap();
+        let config_path = dir.path().join("foo");
+        let config_path = config_path.as_path();
```

Comment:
It's certainly a good way to warm up with these tests. What I'd be interested in seeing is how paths are handled within a git config file.

To my mind, they are indistinguishable from strings, the caller knows if something is supposed to be a path or not.

Thus it's the way a string-like value (I think it's `Other`) here is accessed, it's the callers choice to interpret it as `Path` and the API should help to resolve it as needed.

If this is correct, there would be no `Path` variant in `Value`, as it can't be known in advance.  (I see now why it checks for the existence of a path - it's really to find out if a string value is a path or not. This, however, can't be determined for sure without resolution being available at parse time, not to talk about the IO costs of parsing a config file that way.)

Reply:
Yep, if we can't deal with Path here, it can be bubbled up the call stack. But then the value abstraction is may be abit broken. 

Or we can do like Integer, described above and get `FailedConversion`. The caller then would know its encoding error and can ask for `Other`?

Reply:
> But then the value abstraction is may be abit broken?

I don't know if it's broken, maybe it's something that most users wouldn't need as they usually know what type to expect when querying a value, so no need to 'try' interpreting it as `Value` seems to be doing.

> Or we can do like Integer, described above and get FailedConversion. The caller then would know its encoding error and can ask for Other?

Maybe, to me it all depends on how it's going to be used and for all I know, I'd ask for `path_value(…<context for resolution>)` and it can handle things correctly from there.